### ISSUE TITLE
Darktable doesn't start when building on arch with the newest lensfun

### ISSUE THREAD

**Issue by piratenpanda** [darktable-org/darktable#11248]
I was wondering whether this is a darktable bug or a problem with the arch package which doesn't ship liblensfun.so.2 anymore. 

```
darktable: error while loading shared libraries: liblensfun.so.2: cannot open shared object file: No such file or directory
```

https://archlinux.org/packages/extra/x86_64/lensfun/ ships iblensfun.so, liblensfun.so.0.3.3 and liblensfun.so.1

After symlinking so.1 to so.2 darktable works properly again.

**Comment by piratenpanda** [comment]
0.3.3 has been released a few days ago and is the latest official release. That's why I am confused. Or does 0.3.3 include what was previously 0.3.95?

**Comment by elstoc** [comment]
I wasn't aware of the new release, sorry.

I don't think they've fixed the interoperability issue yet (https://github.com/darktable-org/darktable/issues/2813) so I would still advise using 0.3.2.

**Comment by piratenpanda** [comment]
I think they released it because people might have never run the update command. So I think it's a camera/lens update release yes.

**Comment by elstoc** [comment]
I agree it does look that way. In that case ArchLinux users will likely be in an odd situation where any edits they made on 0.3.2 are now consistent again but any edits made on 0.3.95 will be incorrect on the new version. This will need some testing.

**Comment by parafin** [comment]
> I was wondering whether this is a darktable bug or a problem with the arch package which doesn't ship liblensfun.so.2 anymore.
> 
> ```
> darktable: error while loading shared libraries: liblensfun.so.2: cannot open shared object file: No such file or directory
> ```
> 
> https://archlinux.org/packages/extra/x86_64/lensfun/ ships liblensfun.so, liblensfun.so.0.3.3 and liblensfun.so.1
> 
> After symlinking so.1 to so.2 darktable works properly again. I deleted the opt directory and my build directory before to be sure it's not caused by something else.

Don’t do the symlinking like that, it will cause problems. Instead darktable has to be rebuilt. In any case not darktable issue, consider closing the ticket.

**Comment by piratenpanda** [comment]
> Instead darktable has to be rebuilt.

I tried to, completely from scratch. Does not solve the issue.

**Comment by parafin** [comment]
If you build by yourself, then you probably have lensfun 0.3.95 installed somewhere like /usr/local/. If you're using darktable package from distribution, then it's the one broken and you should report it upstream. Linking errors like that are almost never the fault of software in question, but is due to system misconfiguration.

**Comment by piratenpanda** [comment]
I opened a task. So if any fellow arch user has the same problem, go here https://bugs.archlinux.org/task/74267

**Comment by piratenpanda** [comment]
> If you build by yourself, then you probably have lensfun 0.3.95 installed somewhere like /usr/local/.

I actually forgot that I built it long time ago.. Yeah, works fine now after deleting that and rebuilding darktable. Thanks and sorry for the noise. Much to learn I still have..
